lib/caching-precompiler.js

```diff
@@ -0,0 +1,95 @@
+var cachingTransform = require('caching-transform');
+var fs = require('fs');
+var path = require('path');
+var md5Hex = require('md5-hex');
+var stripBom = require('strip-bom');
+
+module.exports = CachingPrecompiler;
+
+function CachingPrecompiler(cacheDir) {
+	if (!(this instanceof CachingPrecompiler)) {
+		throw new Error('CachingPrecompiler must be called with new');
```

Comment:
@jamestalmage: What we do everywhere else is just invoke ourself with `new` and return it. Maybe switch to that?:

```javascript
if (!(this instanceof CachingPrecompiler)) {
    return new CachingPrecompiler(cacheDir);
}
```